### PR TITLE
feat: Add update shop endpoint 

### LINKED ISSUE #115: Implement shop availability
Owners should have a way to to tell if they are accepting orders or not.

### PR DESCRIPTION
**Pull request by ashrafmahmood**
Fixes #115

### CODE REVIEW

File: alembic/versions/09f4d0a776b0_add_column_is_shop_available.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    #op.drop_table('spatial_ref_sys')
+    op.drop_index('idx_address_location', table_name='address')
+    op.add_column('shops', sa.Column('is_shop_available', sa.Boolean(), nullable=True))
+    op.drop_index('idx_shops_location', table_name='shops')
```
Comment (Haider8) [inline alembic/versions/09f4d0a776b0_add_column_is_shop_available.py]: Comment out `drop_index` statements. We just want to add new column.

**Review by Haider8 — CHANGES_REQUESTED** [review]

File: app/db_models/shop.py [hunk 2]
```diff
@@ -10,6 +10,7 @@ class Shop(Base):
 
     id = Column(Integer, primary_key=True, index=True)
     name = Column(String(length=50), nullable=False)
+    is_shop_available = Column(Boolean, default=True)
```
Comment (Haider8) [inline app/db_models/shop.py]: Just make it `is_available`.

**Review by Haider8 — CHANGES_REQUESTED** [review]

File: alembic/versions/09f4d0a776b0_add_column_is_shop_available.py (hunk 1 above)
Comment (ashrafmahmood) [reply alembic/versions/09f4d0a776b0_add_column_is_shop_available.py]: ok

File: app/db_models/shop.py (hunk 2 above)
Comment (ashrafmahmood) [reply app/db_models/shop.py]: ok

File: app/crud/shop.py [hunk 3]
```diff
@@ -2,8 +2,9 @@
 from sqlalchemy import func
 from sqlalchemy.orm import Session
 
+from app.api.utils.db import clone_db_model
```
Comment (Haider8) [inline app/crud/shop.py]: Remove this

File: app/crud/shop.py [hunk 4]
```diff
@@ -38,3 +39,15 @@ def shop_by_id(db_session: Session, *, shop_id: int) -> Shop:
...
     return None
+
+
+def update_shop(db_session: Session, *, shop: Shop, data: ShopUpdate) -> Shop:
+    data = jsonable_encoder(data, exclude_none=True)
+    for field in data:
+        setattr(shop, field, data[field])
+    if "longitude" in data and "latitude" in data:
+        shop.location = f"POINT({data['longitude']} {data['latitude']})"
+    db_session.add(shop)
```
Comment (Haider8) [inline app/crud/shop.py]: Why are you adding?

File: app/api/api_v1/endpoints/shops.py [hunk 5]
```diff
@@ -68,3 +68,19 @@ async def list_of_shops_for_users(
...
+    shop_id: PositiveInt,
+    db: Session = Depends(get_db),
+    data: ShopUpdate,
+    current_owner: DBOwnerModel = Depends(get_current_owner),
+):
+    if shop := get_shop_by_id(db, shop_id=shop_id, owner_id=current_owner.id):
+        if shop := update_shop(db, shop=shop, data=data):
+            return shop
+    raise HTTPException(
+        status_code=401, detail="This owner cannot update shop with this id."
```
Comment (Haider8) [inline app/api/api_v1/endpoints/shops.py]: `status_code=403`

**Review by Haider8 — CHANGES_REQUESTED** [review]
I think you haven't tested it properly. You are adding new record (with new `id`) in update shop crud.

**Comment by Haider8** [comment]
@ashrafmahmood Please update PR title according to semantic guidelines

File: app/api/api_v1/endpoints/shops.py [hunk 6]
```diff
@@ -68,3 +68,17 @@ async def list_of_shops_for_users(
     if shops:
         return shops
     raise HTTPException(status_code=404, detail="Cannot find shops for your location.")
+
+
+@router.patch("/shops/{shop_id}", response_model=ShopDetails)
```
Comment (Haider8) [inline app/api/api_v1/endpoints/shops.py:73]: Add `is_available` to `ShopDetails` pydantic model

Comment (ashrafmahmood) [reply app/api/api_v1/endpoints/shops.py:73]: ok

**Comment by Haider8** [comment]
Wrong PR title.

**Review by Haider8 — APPROVED** [review]

**Pull request merged**